packages/commonwealth/server/routes/updateChainCategory.ts

```diff
@@ -22,31 +22,28 @@ const updateChainCategory = async (
   res: TypedResponse<UpdateChainCategoryRes>,
   next: NextFunction
 ) => {
-  if (Object.keys(ChainCategoryType).includes(req.body.category)) {
-    const chain = await models.Chain.findOne({
-      where: {
-        id: req.body.chain_id,
-      },
-    });
-    if (!chain) throw new AppError('Invalid Chain Id');
+  const chain = await models.Chain.findOne({
+    where: {
+      id: req.body.chain_id,
+    },
+  });
+  if (!chain) throw new AppError('Invalid Chain Id');
 
-    const existingCategories = chain.category
-      ? (chain.category as string[])
-      : [];
-    if (existingCategories.includes(req.body.category))
-      throw new AppError('Chain already include this category');
-
-    existingCategories.push(req.body.category);
-
-    chain.category = existingCategories.toString();
+  const existingCategories = chain.category ? (chain.category as string[]) : [];
+  const updateCategories = Object.keys(req.body.selected_tags).filter((tag) => {
+    return (
+      req.body.selected_tags[tag] &&
+      Object.keys(ChainCategoryType).includes(tag)
+    );
+  });
+  if (updateCategories != existingCategories) {
```

Comment:
Is this an array equality check? If so, we need to make sure we're doing deep rather than shallow equality.

Reply:
Added deep equal logic, not sure if we had a helper method somewhere but couldnt find one